src/test/java/com/puppycrawl/tools/checkstyle/XpathFileGeneratorAuditListenerTest.java

```diff
@@ -64,7 +64,7 @@ public class XpathFileGeneratorAuditListenerTest {
     private final CloseAndFlushTestByteArrayOutputStream outStream =
             new CloseAndFlushTestByteArrayOutputStream();
 
-    @BeforeAll
+    @BeforeEach
     public static void constructEvents() throws Exception {
```

Comment:
This before is very questionable. It is not modifying any fields and is not doing any assertions. However, removing it results in test failures so obviously there is a pass by reference going on here.

`XpathFileGeneratorAstFilter#accept` is basically populated `XpathFileGeneratorAstFilter#MESSAGE_QUERY_MAP` with pre-filled values to be read in the actual tests.

It is possible to have random test order issues if a new test was added that erased this cache by mistake.

Reply:
Reverted back to `@BeforeAll`

Reply:
IMO, a different way should be done instead of this. Like I said, this cause result in random test order issues which will affect pitest.

Reply:
Initialized them in a different way.

Reply:
Given the design of `XpathFileGeneratorAstFilter`, the only option is to populate the map in a static block, changed to that.